Documentation/Communicational Patterns/2. Tooling/2.1. Tooling.md

```diff
@@ -0,0 +1,127 @@
+![document type: vision](https://themomax.github.io/resources/markdown-labels/document_type_vision.svg)
+
+# Tooling
+
+## Component
+
+A service consists of multiple endpoints. Apodini provides a DSL that allows to define the service's structure using `Component`s. The `Component`-stack is determined on startup and may not change during runtime. `Component`s that provide a `handler` instead of the recursive `content` property define endpoints. Both, input- and output-type of an endpoint are fixed at startup.
```

Comment:
I don't quite get your point. This is exactly the same pattern as we have it right now on the `Component`'s `content: some Component` property and its `handle() -> Response` function.

Reply:
As far as I understand your description and the discussions we had in the last weeks I assumed that `Component`s specify the static structure of the API using the `content: some Component` and do not include the functionality to handle requests. E.g. a `@PathParameter` is only allowed in a `Component` and not in a context where a `handle() -> Response` function exists.
In contrast to `Component`s, `Handler`s are responsible for dynamic behavior and will require some sort of `Layers of Logic` that would be described in a maybe section that might rely on composing `Handler`s or property wrappers. A `Handler` should not include a `content: some Component` but only a `handle() -> Response` method and can include `@Parameter` or `@Environment` property wrappers.
I think we should make this differentiation clear and be sure that this is also defined by the protocols found in `Apodini`.

Reply:
Ah, I think I now see why you are confused: A `Component` cannot have a `func handle()` anymore. It can only have a `var handler: some Handler`. It is just one character difference in the text. Am I right?

Reply:
duplicate